### PR TITLE
Case insensitivity

### PR DESCRIPTION
**Pull request by pond**
Addresses #7, at least to a reasonable degree. There may be edge cases that I haven't considered. The gem's efficiency is significantly decreased by this work, since simple ActiveModel or Ruby Hash operations are no longer possible and instead a degree of data copying, mapping between case-normalized and actual-case forms, and method introspection becomes necessary. The impact is hopefully not too severe in practice - it'll depend upon the use cases of the client Rails application.

**View diff with white space ignored** else the diff size, especially in tests, will be unnecessarily large. Quite a few tests have been nested inside a `shared_examples` declaration so that they can be run twice, once with attributes in schema-normal case and once in uppercase, to verify that case-insensitive behaviour is functioning as intended.

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.7.2
+2.7.4
```
Comment (pond) [inline .ruby-version:1]: There's some maintenance on Gem and Ruby versions collected up in passing here.

File: app/models/scimitar/resources/mixin.rb [hunk 2]
```diff
@@ -542,20 +542,20 @@ def to_scim_backend(data_source:, attrs_map_or_leaf_value:)
           #   ::scim_attributes_map.
           #
           #     {                                                     | {
-          #       "userName": "foo",                                  |   "id": "id",
-          #       "name": {                                           |   "externalId": :scim_uid",
-          #         "givenName": "Foo",                               |   "userName": :username",
-          #         "familyName": "Bar"                               |   "name": {
-          #       },                                                  |     "givenName": :first_name",
-          #       "active": true,                                     |     "familyName": :last_name"
+          #       "userName": "foo",                                  |   'id': :id,
```
Comment (pond) [inline app/models/scimitar/resources/mixin.rb:545]: Drive-by fixes on would-be-syntax errors in comment.

File: spec/models/scimitar/resources/mixin_spec.rb [hunk 3]
```diff
@@ -677,18 +693,18 @@ def self.scim_timestamps_map
...
                     ]
-                  }
+                  }.with_indifferent_case_insensitive_access()
 
                   @instance.send(
                     :from_patch_backend!,
                     nature:        'add',
                     path:          path,
-                    value:         'added_over_origina@test.com',
+                    value:         'added_over_original@test.com',
```
Comment (pond) [inline spec/models/scimitar/resources/mixin_spec.rb:702]: Includes drive-by fix of minor but annoying typo which was replicated a lot by copy-and-paste.

**Review by akampjes — APPROVED** [review]
